src/kubernetes.cc

```diff
@@ -529,6 +535,54 @@ const std::string KubernetesReader::ClusterFullName() const {
                              {"cluster_name", cluster_name}});
 }
 
+const std::pair<std::string, std::string> KubernetesReader::TypeAndVersion(
+    const std::string& api_version, const std::string& kind) const {
+  size_t num_slashes = std::count(api_version.begin(), api_version.end(), '/');
+  if (num_slashes == 0) {
+    // api_version has the format "<version>". E.g. "v1".
+    const std::string resource_type =
+        format::Substitute(kK8sCoreGroupResourceTypeFormat, {{"kind", kind}});
+    return std::make_pair(resource_type, api_version);
+  } else if (num_slashes == 1) {
+    // api_version has the format "<group-name>/<version>". E.g. "apps/v1".
+    size_t slash_position = api_version.find('/');
+    const std::string group_name = api_version.substr(0, slash_position);
+    const std::string version =
+        api_version.substr(slash_position + 1, std::string::npos);
+    const std::string resource_type =
+        format::Substitute(kK8sNamedGroupsResourceTypeFormat,
+                           {{"group_name", group_name}, {"kind", kind}});
+    return std::make_pair(resource_type, version);
+  } else {
+      LOG(ERROR) << "Invalid apiVersion: " << api_version;
+  }
+}
+
+const std::string KubernetesReader::FullResourceName(
+    const std::string& self_link) const {
+  std::vector<std::string> slash_split;
+  boost::algorithm::split(
+      slash_split, self_link, boost::algorithm::is_any_of("/"));
+
+  std::vector<std::string> link_components;
+  if(slash_split[1] == "api") {
+    // Core resources, start with "/api/<version>/..."
+    link_components.assign(slash_split.begin() + 3, slash_split.end());
```

Comment:
Is the goal here to exclude version?

Reply:
Correct.

Reply:
Can you please add a comment that calls this out?

Reply:
See the updated code - I have added a comment.